tests/compile-fail/methods.rs

```diff
@@ -355,3 +355,92 @@ fn clone_on_double_ref() {
                                 //~^^^ERROR using `clone` on a `Copy` type
     println!("{:p} {:p}",*y, z);
 }
+
+fn single_char_pattern() {
+    let x = "foo";
+    x.split("x");
+    //~^ ERROR single-character string constant used as pattern
+    //~| HELP try using a char instead:
+    //~| SUGGESTION x.split('x');
+
+    x.split("xx");
+
+    x.split('x');
+
+    let y = "x";
+    x.split(y);
+
+    // Not yet testing for multi-byte characters
+    // Changing `r.len() == 1` to `r.chars().count() == 1` in `lint_single_char_pattern`
+    // should have done this but produced an ICE
```

Comment:
I couldn't find one, although there may be (there are a lot of issues on rustc to filter through). It seems to happen here:

    tests/compile-fail/methods.rs:377:13: 377:16 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#single_char_pattern
    tests/compile-fail/methods.rs:378:13: 378:16 error: single-character string constant used as pattern
    tests/compile-fail/methods.rs:378     x.split("💣");
                                                  ^~~
    tests/compile-fail/methods.rs:4:9: 4:15 note: lint level defined here
    tests/compile-fail/methods.rs:4 #![deny(clippy, clippy_pedantic)]
                                            ^~~~~~
    tests/compile-fail/methods.rs:378:5: 378:17 help: try using a char instead:
    error: internal compiler error: Error constructed but not emitted
    thread panicked while panicking. aborting.

Probably related, the formatting on the hint output gets broken on multibyte chars, e.g.:

    tests/compile-fail/methods.rs:377:13: 377:16 error: single-character string constant used as pattern
    tests/compile-fail/methods.rs:377     x.split("ℝ");
                                                  ^~~
    tests/compile-fail/methods.rs:4:9: 4:15 note: lint level defined here
    tests/compile-fail/methods.rs:4 #![deny(clippy, clippy_pedantic)]
                                            ^~~~~~
    tests/compile-fail/methods.rs:377:5: 377:17 help: try using a char instead:
    tests/compile-fail/methods.rs:        x.split('ℝ')");